### DISCUSSION TITLE
How to Store Memory for Context Maintenance in a Chat Session? (Autogen Agent - Sequential Flow)

### DISCUSSION THREAD

**Discussion by Priya-Issar** [mem0ai/mem0#2051] [Q&A]
I’m working on building a Text-to-SQL LLM chatbot that takes user questions from the frontend and generates SQL queries as output.

Architecture(using Autogen Sequential Flow):
Here’s the detailed workflow I’m using:-
1. User Proxy Agent: Takes the user question from the frontend.
2. Assistant Agent: Understands the use case and prepares a task plan to solve the query.
3. Assistant Agent: Retrieves the DDL from the vector database based on the user query (text question from the User Proxy Agent).
4. Assistant Agent: Uses the retrieved DDL to generate the SQL query.

Goal:
I want to store memory to maintain the context throughout a single chat session.

Questions:
1. What’s the best way to integrate memory storage into this workflow to maintain context?
2. Are there specific strategies for efficiently storing and retrieving chat session data for such a use case?
3. If anyone has sample code or repositories, could you kindly share them for reference?
Looking forward to suggestions.

**Comment by hrishit123** [comment]
Sorry to say but thats impossible.
